src/java.base/share/classes/sun/launcher/LauncherHelper.java

```diff
@@ -820,12 +820,19 @@ private static Class<?> loadModuleMainClass(String what) {
     private static Class<?> loadMainClass(int mode, String what) {
         // get the class name
         String cn;
+        // store the jar file
```

Comment:
"store the jar file" is a confusing comment to add. I think what you want to say is that the JarFile will put the underlying file in the JarFile/ZipFile cache and this will avoid needing to re-parse the manifest when the JAR file is opened on the class path, triggered by Class.forName.

Reply:
Updating to use the precise description.